### PR TITLE
add snapshot repo envs

### PR DESCRIPTION
**Pull request by thepsalmist**
Add Elasticsearch snapshot repo env variables to deploy.sh

### CODE REVIEW

File: docker/deploy.sh [hunk 1]
```diff
@@ -347,6 +349,7 @@ dev)
     ELASTICSEARCH_CONTAINERS=1
     ELASTICSEARCH_SHARD_COUNT=2
     ELASTICSEARCH_SHARD_REPLICAS=1
+    ELASTICSEARCH_SNAPSHOT_REPO=test-mediacloud-elasticsearch-snapshots
```
Comment (kilemensi) [inline docker/deploy.sh:352]: Isn't this dev? Aren't things deployed per $LOGIN_USER?

File: docker/deploy.sh [hunk 2]
```diff
@@ -296,6 +296,7 @@ prod)
     # ES index settings are static, prod settings should not change
     ELASTICSEARCH_SHARD_COUNT=30
     ELASTICSEARCH_SHARD_REPLICAS=1
+    ELASTICSEARCH_SNAPSHOT_REPO=mediacloud-elasticsearch-snapshots
```
[the remark this answers is not in the document]
Comment (thepsalmist) [reply docker/deploy.sh:299]: @kilemensi when creating the ES repo's we followed the scheme of the othe S3 buckets named as `mediacloud-XXX`

File: docker/deploy.sh (hunk 1 above)
Comment (philbudne) [reply docker/deploy.sh:352]: For the importer, the staging S3 bucket is mediacloud-staging-indexer-archive (keeping mediacloud in front).
Development deployments don't archive to S3.

(How) is this going to work for a developer deploy?  There's no private config file for development deploys.

File: indexer/scripts/elastic-conf.py [hunk 3]
```diff
@@ -75,7 +75,9 @@ def process_args(self) -> None:
         self.replicas = self.args.replicas
         self.ilm_max_age = self.args.ilm_max_age
         self.ilm_max_shard_size = self.args.ilm_max_shard_size
-        self.es_snapshot_repo = self.args.es_snapshot_repo
+        self.es_snapshot_repo = getattr(
```
Comment (philbudne) [inline indexer/scripts/elastic-conf.py:78]: Why is getattr needed?

File: docker/deploy.sh [hunk 4]
```diff
@@ -61,7 +61,7 @@ usage() {
 PIPELINE_TYPE=queue-fetcher	# default (2024-04-22)
 
 # take command line option? used for input and archive output (if created)
-HIST_YEAR=2023
+HIST_YEAR=2022
```
[the remark this answers is not in the document]
Comment (kilemensi) [reply docker/deploy.sh:64]: FYI: https://github.com/mediacloud/story-indexer/pull/289 implements `HIST_YEAR` as a command line option so we no longer need this change.

**Comment by thepsalmist** [comment]
PS, we're adding a `mediacloud-test-elasticsearch-snapshots` repo to the dev deployment stage, since [Elasticsearch Snapshot AP](https://www.elastic.co/guide/en/elasticsearch/reference/current/create-snapshot-api.html#create-snapshot-api-path-params)I does not allow a `blank/None` repository name, required for `elastic-conf`. Tracking issue here #290

File: indexer/scripts/elastic-conf.py [hunk 5]
```diff
@@ -165,7 +171,7 @@ def create_slm_policy(self, es: Elasticsearch) -> Any:
         schedule = json_data["schedule"]
         retention = json_data["retention"]
 
-        # SLM starts automatically when clusture is formed.If SLM stopped start using POST /_slm/start
+        # SLM starts automatically when cluster is formed.If SLM stopped start using POST /_slm/start
```
Comment (kilemensi) [inline indexer/scripts/elastic-conf.py:174]: We probably don't need this comment... should be part of docs if really important.

**Comment by thepsalmist** [comment]
@philbudne any further concerns on this?

**Review by philbudne — APPROVED** [review]

**Pull request merged**